pkg/providers/tinkerbell/validator.go

```diff
@@ -94,6 +102,14 @@ func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, tinkerbel
 	return nil
 }
 
+func (v *Validator) validateControlPlaneIpUniqueness(tinkerBellClusterSpec *spec) error {
```

Comment:
Should we move validateControlPlaneIpUniqueness to one common place from where each provider can use the same validation if possible?